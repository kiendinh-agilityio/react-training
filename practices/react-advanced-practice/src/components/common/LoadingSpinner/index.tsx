// Import radix ui
import { Box } from '@radix-ui/themes';

// Import styles animation for toast
import './loading.css';

const LoadingSpinner = () => (
  <Box className="loading-container">
    <Box as="span" className="loading-indicator"></Box>
  </Box>
);

export default LoadingSpinner;
