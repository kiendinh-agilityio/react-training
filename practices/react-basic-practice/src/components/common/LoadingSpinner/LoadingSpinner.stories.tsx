import { Meta, StoryObj } from '@storybook/react';
import LoadingSpinner from '.';

// Define metadata for the story
const meta: Meta<typeof LoadingSpinner> = {
  title: 'LoadingSpinner',
  component: LoadingSpinner,
  decorators: [
    (Story) => (
      <div className="relative w-full h-screen flex items-center justify-center">
        <Story />
      </div>
    ),
  ],
  parameters: {
    layout: 'fullscreen',
  },
};

export default meta;

// Story type definition
type Story = StoryObj<typeof LoadingSpinner>;

export const Loading: Story = {
  render: () => <LoadingSpinner />,
};
