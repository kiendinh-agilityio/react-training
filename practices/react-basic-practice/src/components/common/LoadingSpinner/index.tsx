// Import styles animation for toast
import './loading.css';

const LoadingSpinner = () => (
  <div className="loading-container">
    <span className="loading-indicator"></span>
  </div>
);

export default LoadingSpinner;
