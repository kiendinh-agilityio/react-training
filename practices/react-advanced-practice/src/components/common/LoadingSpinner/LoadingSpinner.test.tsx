import { render } from '@testing-library/react';
import LoadingSpinner from '.';

describe('LoadingSpinner component', () => {
  it('matches the snapshot', () => {
    const { container } = render(<LoadingSpinner />);

    expect(container).toMatchSnapshot();
  });
});
