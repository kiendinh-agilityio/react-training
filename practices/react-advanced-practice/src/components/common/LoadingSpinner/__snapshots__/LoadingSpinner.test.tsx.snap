// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSpinner component matches the snapshot 1`] = `
<div>
  <div
    class="rt-Box loading-container"
  >
    <span
      class="rt-Box loading-indicator"
    />
  </div>
</div>
`;
